src/util/bson/mongo_timestamp.hpp

```diff
@@ -19,39 +19,27 @@
 #ifndef REALM_BSON_DATETIME_HPP
 #define REALM_BSON_DATETIME_HPP
 
-#include <ctime>
+#include <cstdint>
 
 namespace realm {
 namespace bson {
 
 struct MongoTimestamp {
+    MongoTimestamp(uint32_t seconds, uint32_t increment) :seconds(seconds), increment(increment) {}
 
-    MongoTimestamp(const int64_t seconds, const int64_t increment);
-    
-    int64_t seconds_since_epoch() const
+    friend bool inline operator==(const MongoTimestamp& lhs, const MongoTimestamp& rhs)
```

Comment:
This is the currently recommended way to define these operators. They are called "hidden friends", declared as friends inside the class, and never declared outside of the class, at least not in headers. The main advantage over declaring the outside of the class is that the compiler won't look to see if there is any conversion to MongoTimestamp every time it tries to resolve a usage of `==`. If you've ever seen rediculously long lists of `operator==` functions whenever you have a typo that makes it not resolve, that is because it looks at Every Single One to see if it matches and lists them when none do. This technique keeps is out of that list.

Reply:
👍